sdk/src/asset_handlers/png_io.rs

```diff
@@ -477,15 +478,18 @@ impl AssetIO for PngIO {
             .open(asset_path)
             .map_err(Error::IoError)?;
 
-        let mut output_stream = std::fs::OpenOptions::new()
-            .read(true)
-            .write(true)
-            .open(asset_path)
-            .map_err(Error::IoError)?;
+        let mut temp_file = Builder::new()
```

Comment:
Why are we going to temp files at this level? Isn't it better to let the higher levels pass in a temp file?

Reply:
If you look at this api it only takes a destination.  If you fail you have corrupted the source.